### PR TITLE
Adding e2e Test Template for Module To Module.

### PR DESCRIPTION
**Pull request by aribeironovaes**

### CODE REVIEW

File: smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      }
+    },
+    "$edgeHub": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "routes": {
+          "TempFilterToCloud": "FROM /messages/modules/tempFilter/outputs/* INTO $upstream",
+          "TempSensorToTempFilter": "FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/tempFilter/inputs/input1\")"
+
+          "upstreamLoadGen1": "FROM /messages/modules/loadGen1/outputs/output1 INTO $upstream",
```
Comment (varunpuranik) [inline smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: Don't think we need the loadGen routes?

Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: Oops. Though I have removed. Thanks for catching.

Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: Fixed.

File: smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          },
+          
+        }
+      }
+    },
+    "$edgeHub": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "routes": {
+          "TempFilterToCloud": "FROM /messages/modules/tempFilter/outputs/* INTO $upstream",
```
Comment (varunpuranik) [inline smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: Should the routes be something that can be configured? What if different tests need different routes?

Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: The idea is to have 1 deployment per test. (we have the default one for TempSensor test).

Comment (varunpuranik) [reply smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: Right, but the template should be across tests, right?

Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_module_deployment.template.json]: I though about that, but that would require our test infraestructure (aka. Quick Start) to be a lot different than today. I agree we can improve a lot in our e2e test. But as of now each E2E Test case will be 1 VSTS task, which will require a full deployment. So, having a single deployment template and changing dynamically would be a lot more complex. So, the name (template) on the file just means that it has parameters (build number and credentials) that needs to be filled from Script. This is very similar to what we are doing today on Stress/Long Haul test. 

Ideally, we should be using a test framework, where we can have 1 single Setup, 1 single tead down and each test case would be just a set of TestSetup (changing deployment)  and a Test Verify... 

---
In reply to: [209095732](https://github.com/Azure/iotedge/pull/140#discussion_r209095732) [](ancestors = 209095732)

**Pull request merged**